rig/place_and_route/place/sa/c_kernel.py

```diff
@@ -0,0 +1,129 @@
+"""Wraps the "rig_c_sa" Python C extension library in a Kernel object."""
+
+from collections import defaultdict
+
+from itertools import chain, repeat
+
+from six import iteritems
+
+import pkg_resources
+
+# An optional Rig dependency.
+from rig_c_sa import ffi
+import rig_c_sa
+
+
+# Make sure installed rig_c_sa version is compatible.
+pkg_resources.require("rig_c_sa>=0.3.1,<1.0.0")
+
+
+class CKernel(object):  # pragma: no cover
+    """An implementation of the Simulated Annealing placement algorithm kernel
+    written in C.
+
+    Thanks to being written in C, this kernel is 50-150x faster than
+    :py:class:`~rig.place_and_route.place.sa.python_kernel.PythonKernel` while
+    maintaining the same placement quality. Since installing the C
+    kernel requires a compiler it is not included in the standard Rig package.
+    The C kernel can be installed using::
+
+        $ pip install rig_c_sa
+
+    The `rig_c_sa <https://github.com/project-rig/rig_c_sa>`_ package is a
+    Python/C library containing the C components of the algorithm. Installation
+    requires that you have C compiler installed along with `libffi
+    <https://sourceware.org/libffi/>`_ which are readily available in the
+    package managers of real operating systems.
```

Comment:
I think we should try not to bash Windows too much.

(Otherwise we look like a document I recently read on routing table compression which got very tiring after a while and all the "Micro$ofts"...)